homedir.go

```diff
@@ -81,9 +82,28 @@ func dirUnix() (string, error) {
 		return home, nil
 	}
 
-	// If that fails, try the shell
+	// If that fails, try getent
 	var stdout bytes.Buffer
-	cmd := exec.Command("sh", "-c", "eval echo ~$USER")
+	cmd := exec.Command("getent", "passwd", strconv.Itoa(os.Getuid()))
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		// If "getent" is missing, ignore it
+		if err != exec.ErrNotFound {
+			return "", err
+		}
+	} else {
+		if passwd := strings.TrimSpace(stdout.String()); passwd != "" {
+			// username:password:uid:gid:gecos:home:shell
+			passwdParts := strings.SplitN(passwd, ":", 7)
+			if len(passwdParts) > 5 {
+				return passwdParts[5], nil
+			}
+		}
+	}
+
+	// If all else fails, try the shell
+	stdout.Reset()
+	cmd = exec.Command("sh", "-c", "cd 2>/dev/null; pwd")
```

Comment:
Used `2>/dev/null` and `;` here so that if the `cd` fails to discover the current user's home directory, we can just ignore the failure and still get our current directory instead.  If that's not the desired behavior, this is easy to swap to just use `cd && pwd` so that it'll outright fail and return an `error` instead. :+1: